src/cargo/core/compiler/context/unit_dependencies.rs

```diff
@@ -38,27 +50,33 @@ fn deps_of<'a, 'b, 'cfg>(
     unit: &Unit<'a>,
     cx: &Context<'a, 'cfg>,
     deps: &'b mut HashMap<Unit<'a>, Vec<Unit<'a>>>,
+    profile_for: ProfileFor,
```

Comment:
I feel uneasy about this `profile_for` argument. This function is memoized, so the sequence of calls `deps_of(u, cx, deps, ProfileFor::Any); deps_of(u, cx, deps, ProfileFor::TestDependency)` will return the same result each time (and the result might be different if we swap the order of calls).

Reply:
I understand what you mean.  In practice, that's never a problem for `TestDependency` since it only exists for clearing `panic`, and you'll never ask for a unit *with* panic set as a `TestDependency`.  

I'm trying to think of a scenario where it would matter for `CustomBuild`, but I can't think of any.  It might matter if we later add a "build-script"-specific profile, and you have an override deep in a dependency chain.

Adding it to the map would require a separate de-duplication phase, which I think would add ~15 lines of code.

Let me know if you think it's important.  I'd lean towards not changing it unless there's a testable case.  Or to make it easier to add a build-script profile later if there's a strong chance of that.

Reply:
I think I'm fine with it: I would love to refactor code in such a way that this problem just not arises, or at least to put some sort of run-time assertion, but I have no ideas of how to actually do this. Storing pairs in the `HashMap` would be worse than the status quo. 

Let's perhaps just add a small comment here though?